vllm/model_executor/layers/rotary_embedding.py

```diff
@@ -109,7 +109,7 @@ def _forward(
             key_pass = key[..., self.rotary_dim:]
 
         self.cos_sin_cache: torch.Tensor = self.cos_sin_cache.to(
-            positions.device)
+            positions.device, dtype=query.dtype)
```

Comment:
This was a bug. we always use the default dtype, but the kernel requires it to be the same as query dtype. this breaks the kernel because cos_sin_cache dtype is sometimes float16 (default dtype) while query dtype was bfloat16

Reply:
I'm even surprised this is called instead of the compiled kernel